modules/data-entry/src/main/java/ca/sickkids/ccm/lfs/QueryBuilder.java

```diff
@@ -167,25 +208,55 @@ private String fullTextEscape(String input)
      */
     private Iterator<Resource> quickSearch(String query) throws RepositoryException
     {
-        final String[] toSearch = {"lfs:Form", "lfs:Subject", "lfs:Questionnaire"};
-        final StringBuilder oakQuery = new StringBuilder();
-
-        for (int i = 0; i < toSearch.length; i++) {
-            oakQuery.append(
-                String.format(
-                    "select n.* from [%s] as n where contains(*, '*%s*')",
-                    toSearch[i],
-                    this.fullTextEscape(query)
-                )
-            );
-
-            // Union interstitial terms together
-            if (i + 1 != toSearch.length) {
-                oakQuery.append(" union ");
+        ArrayList<Resource> outputList = new ArrayList<Resource>();
+
+        final StringBuilder xpathQuery = new StringBuilder();
+        xpathQuery.append("/jcr:root/Forms//*[jcr:like(fn:lower-case(@value),'%");
+        xpathQuery.append(this.fullTextEscape(query.toLowerCase()));
+        xpathQuery.append("%'");
+        xpathQuery.append(" )]");
+
+        Iterator<Resource> foundResources = queryXPATH(xpathQuery.toString());
+        /*
+        * For each Resource in foundResources, move up the tree until
+        * we find an ancestor node of type `lfs:Form`
+        */
+        while (foundResources.hasNext()) {
+            Resource thisResource = foundResources.next();
+            Resource thisParent = thisResource;
+            String[] resourceValues = thisResource.getValueMap().get("value", String[].class);
+            while (thisParent != null && !"lfs/Form".equals(thisParent.getResourceType())) {
+                thisParent = thisParent.getParent();
+            }
+            if (thisParent != null) {
+                String resourceValue = getMatchingFromArray(resourceValues, query);
```

Comment:
```suggestion
            String[] resourceValues = thisResource.getValueMap().get("value", String[].class);
            String resourceValue = getMatchingFromArray(resourceValues, query);
            while (thisParent != null && !"lfs/Form".equals(thisParent.getResourceType())) {
                thisParent = thisParent.getParent();
            }
            if (thisParent != null && resourceValue != null) {
```
Guard against `getMatchingFromArray` returning `null` by moving the resourceValue outside of the conditional and ensuring it is not null (e.g. when the search was found using wildcard characters)